x-pack/plugins/alerting/server/routes/lib/rewrite_actions.ts

```diff
@@ -30,3 +30,20 @@ export const rewriteActions: (
       } as RuleAction)
   );
 };
+
+export const rewriteActionsRes = (actions?: RuleAction[]) => {
+  const rewriteFrequency = ({ notifyWhen, ...rest }: NonNullable<RuleAction['frequency']>) =>
+    ({
+      ...rest,
+      notify_when: notifyWhen,
+    } as ReqRuleAction['frequency']);
+  if (!actions) return [];
+  return actions.map(
+    ({ actionTypeId, ...action }) =>
+      ({
+        ...action,
+        connector_type_id: actionTypeId,
+        ...(action.frequency ? { frequency: rewriteFrequency(action.frequency) } : {}),
+      } as ReqRuleAction & { connector_type_id: RuleAction['actionTypeId'] })
```

Comment:
Why do we need casting here?

Reply:
Lol idk ask the Typescript parser, I couldn't figure out any other way to make it happy

Reply:
I've asked because if I delete these casting in your branch, it do not turn red. Maybe linter behaves weird from my side(it happens sometimes). Can you double check, please.